webapp/app/viewModels/role/RoleDelete.js

```diff
@@ -0,0 +1,23 @@
+require('services/roleService');
```

Comment:
I think you don't need a new page+vm in order to delete a role. Since you already have a role list you can just make the call to the API from a button in that page